src/auframe/auframe.c

```diff
@@ -76,3 +81,33 @@ void auframe_mute(struct auframe *af)
 
 	memset(af->sampv, 0, auframe_size(af));
 }
+
+
+/**
+ * Computes audio frame volume and compares with a boundary to decide if it is
+ * nearly silence
+ *
+ * @param af  Audio frame
+ *
+ * @return true if the frame is nearly silence
+ */
+bool auframe_silence(const struct auframe *af)
+{
+	const int16_t *v;
+	int32_t sum = 0;
+	size_t i;
+
+	if (af->fmt != AUFMT_S16LE)
+		return true;
+
+	v = af->sampv;
+
+	for (i = 0; i < af->sampc; i++) {
+		sum += v[i]*v[i];
+
+		if (sum > (int32_t) (i + 1) * SILENCE_Q)
```

Comment:
I wounder if we should use `aulevel_calc_dbov` and safe this value in af->level for later re-use. Should I prepare
a PR with something like this?
```c
double auframe_level(struct auframe *af) {
  if (!af->level)
    af->level = aulevel_calc_dbov(...);

   return af->level;
}
```

Reply:
This would be nice. Then the silence detection can be removed from the PR.